### PR TITLE
NMA-800  Strandhogg Vulnerability v1 and v2

### PR DESCRIPTION
**Pull request by tomasz-ludek**
## Issue being fixed or feature implemented

NMA-800 Strandhogg Vulnerability v1 and v2
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

**Comment by HashEngineering** [comment]
found this exception after Security / Reset Wallet:

```
2021-03-26 09:34:18.048 27547-27547/hashengineering.darkcoin.wallet E/AndroidRuntime: FATAL EXCEPTION: main
    Process: hashengineering.darkcoin.wallet, PID: 27547
    java.lang.RuntimeException: Unable to start activity ComponentInfo{hashengineering.darkcoin.wallet/de.schildbach.wallet.ui.MoreActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'org.bitcoinj.core.Coin org.bitcoinj.wallet.Wallet.getBalance(org.bitcoinj.wallet.Wallet$BalanceType)' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'org.bitcoinj.core.Coin org.bitcoinj.wallet.Wallet.getBalance(org.bitcoinj.wallet.Wallet$BalanceType)' on a null object reference
        at de.schildbach.wallet.ui.LockScreenActivity.setupBackupSeedReminder(LockScreenActivity.kt:137)
        at de.schildbach.wallet.ui.LockScreenActivity.onCreate(LockScreenActivity.kt:101)
        at de.schildbach.wallet.ui.MoreActivity.onCreate(MoreActivity.kt:38)
        at android.app.Activity.performCreate(Activity.java:7009)
        at android.app.Activity.performCreate(Activity.java:7000)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856) 
        at android.app.ActivityThread.-wrap11(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6494) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
```

File: wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt [hunk 1]
```diff
@@ -110,12 +114,13 @@ class OnboardingActivity : RestoreFromFileActivity() {
...
-        } else {
-            WalletActivity.createIntent(this)
-        }
-        startActivity(intent)
+//        val intent = if (walletApplication.configuration.autoLogoutEnabled) {
+//            LockScreenActivity.createIntent(this)
+//        } else {
+//            WalletActivity.createIntent(this)
+//        }
+//        startActivity(intent)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt:122]: Here is some unneeded comments

File: wallet/src/de/schildbach/wallet/ui/TransactionResultActivity.kt [hunk 2]
```diff
@@ -91,7 +91,8 @@ class TransactionResultActivity : AbstractWalletActivity() {
                         startActivity(WalletActivity.createIntent(this))
                     }
                     else -> {
-                        startActivity(LockScreenActivity.createIntentAsNewTask(this))
+//                        startActivity(LockScreenActivity.createIntentAsNewTask(this))
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/TransactionResultActivity.kt:94]: comment

File: wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt [hunk 3]
```diff
@@ -45,14 +45,18 @@ class OnboardingActivity : RestoreFromFileActivity() {
         @JvmStatic
         fun createIntent(context: Context): Intent {
             return Intent(context, OnboardingActivity::class.java)
-                    .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+//                    .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt:48]: comment

Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt:48]: Fixed

File: wallet/src/de/schildbach/wallet/ui/TransactionResultActivity.kt (hunk 2 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/TransactionResultActivity.kt:94]: Fixed

File: wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt (hunk 1 above)
Comment (tomasz-ludek) [reply wallet/src/de/schildbach/wallet/ui/OnboardingActivity.kt:122]: Fixed

**Pull request merged**